src/components/lib/fetch-data.ts

```diff
@@ -0,0 +1,31 @@
+import { createPublicClient, http } from 'viem';
+import { optimism } from 'viem/chains'
+
+const client = createPublicClient({
```

Comment:
Use `l2PublicClient` here from `@/lib/chains.ts` instead of creating a new client.

Reply:
fixed